Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1

```diff
@@ -106,15 +105,150 @@ Describe 'Identity Cli Functional Tests' {
     Describe 'Invoke-UpdateClientPassword' {}
     
     Describe 'Test-IsClientRegistered' {}
-    
-    Describe 'Get-ApiRegistration' {}
-    
-    Describe 'New-ApiRegistration' {}
-    
-    Describe 'Invoke-UpdateApiRegistration' {}
-    
-    Describe 'Invoke-UpdateClientPassword' {}
-    
-    Describe 'Test-IsApiRegistered' {}
 
+    Describe 'Api Registration Functional Tests' -tag "Functional" {
+	    Context 'Api Registration Scenarios' {
+            	# Get Access token from identity to pass to registration
+			    $accessToken = Get-AccessToken -identityUrl $identityUrl -clientId "fabric-installer" -secret $installerSecret -scope "fabric/identity.manageresources"
+
+                # Get-FabricInstallerAccessToken returns an access token if request succeeds, expect a value when successful
+                $accessToken | Should -Not -Be $null
+
+                # New-ApiRegistrationBody returns an apiresource object
+                $newApiResource = New-ApiRegistrationBody `
+                -apiName "test-Api" `
+                -scopes @{"name" = "test-Api"; "displayName" = "Test-API"} `
+                -userClaims @("name", "email", "role", "groups") `
+                -isEnabled true
+
+                $jsonApi = $newApiResource | ConvertTo-Json
+
+            It 'Should not fail to Test, Create, Test, Get, Edit and Delete Registration' {
+
+                # Test an api not registered
+                $testApi = Test-IsApiRegistered -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $testApi | Should -Be $false
+
+                # Create an api not registered
+                $newApi = New-ApiRegistration -identityUrl $identityUrl -body $jsonApi -accessToken $accessToken
+
+                $newApi | Should -Not -Be $null
+
+                # Test an api that is registered
+                $testApi = Test-IsApiRegistered -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $testApi | Should -Be $true
+
+                # Get an api that is registered
+                $newApi = Get-ApiRegistration -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $newApi | Should -Not -Be $null
```

Comment:
Should this also check to see if it returns some values that are expected?

Reply:
good catch